### ISSUE TITLE
Simplify monoT5 and monoBERT boilerplate

### ISSUE THREAD

**Issue by lintool** [castorini/pygaggle#80]
There's a lot of boilerplate here: https://github.com/castorini/pygaggle#a-simple-reranking-example

Can we fold all of that into the constructor of the class? E.g., so we're left with:

```python
reranker =  monoT5()
```

or

```python
reranker =  monoBERT()
```

Make `model_name`, `tokenizer_name`, etc.  configurable with sensical defaults.

So simple reranking gets boiled down to

```python
from pyserini.search import SimpleSearcher
from pygaggle.rerank.base import hits_to_texts

query = Query('who proposed the geocentric theory')
searcher = SimpleSearcher('/path/to/msmarco/index/')
reranker = monoBERT()

hits = searcher.search(query.text)
reranked = reranker.rerank(query, hits_to_texts(hits))
reranked.sort(key=lambda x: x.score, reverse=True)
```

@rodrigonogueira4 @rodrigonogueira4 thoughts?

**Comment by ronakice** [comment]
Yeah, I think we should do that!

**Comment by yuxuan-ji** [comment]
I'm a bit confused by the naming of this, is the goal to have a set of predefined rerankers w/ defaults
ex:
```python
from pyaggle.rerank.pretrained import monoBERT, monoT5
reranker = monoBERT() or monoT5() -> Reranker # defaults to castorini/monobert-large-msmarco and castorini/monot5-base-msmarco

# similar to how huggingface transformers has:
AutoModel.from_pretrained('monoBERT')
```

or just to create general constructors for each reranker similar to what's in:
https://github.com/castorini/pygaggle/blob/master/pygaggle/run/evaluate_document_ranker.py#L103
i.e.
```python
def construct_seq_class_transformer(options: DocumentRankingEvaluationOptions
                                    ) -> Reranker:
    model = AutoModelForSequenceClassification.from_pretrained(options.model, from_tf=options.from_tf)
    device = torch.device(options.device)
    model = model.to(device).eval()
    tokenizer = AutoTokenizer.from_pretrained(options.tokenizer_name)
    return SequenceClassificationTransformerReranker(model, tokenizer)
```

TLDR why name `monoBERT` vs `construct_seq_class_transformer`
